action.yaml

```
@@ -0,0 +1,51 @@
+name: Elixir Build and Cache Dependencies
+inputs:
+  elixir-version:
+    required: true
+    type: string
+  erlang-version:
+    required: true
+    type: string
+  mix-env:
+    required: true
+    type: string
+outputs:
+  cache-key:
+    value: ${{ steps.compute-cache-key.outputs.cache-key }}
+  cache-key-suffix:
+    value: ${{ steps.compute-cache-key.outputs.cache-key-suffix }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: erlef/setup-elixir@v1
+      with:
+        elixir-version: ${{ inputs.elixir-version }}
+        otp-version: ${{ inputs.erlang-version }}
+    - name: Compute Cache Key
+      id: compute-cache-key
+      run: |
+        elixir="elixir-${{ inputs.elixir-version }}"
+        erlang="erlang-${{ inputs.erlang-version }}"
+        mix_env="${{ inputs.mix-env }}"
+        runner="${{ runner.os }}"
+        suffix="${runner}_${erlang}_${elixir}_${mix_env}_${{ hashFiles('mix.lock') }}"
+        result="cache-deps_${suffix}"
+        echo "::set-output name=cache-key::$result"
+        echo "::set-output name=cache-key-suffix::$suffix"
+      shell: bash
+    - id: cache-dependencies
+      uses: actions/cache@v2
+      with:
+        key: ${{ steps.compute-cache-key.outputs.cache-key }}
+        path: |
+          deps
+          _build
```

The [Elixir example](https://github.com/actions/cache/blob/main/examples.md#elixir---mix) for `actions/cache` includes `restore-keys` option. Unfortunately, the documentation around it doesn't elaborate much. I assume the code as you have it is fine, but it might be something to play with to see if it helps performance.
```suggestion
    - id: cache-dependencies
      uses: actions/cache@v2
      with:
        key: ${{ steps.compute-cache-key.outputs.cache-key }}
        path: |
          deps
          _build
        restore-keys: cache-deps_
```

---

Restoring partial cache keys in this case might include caches from a different OS, Elixir/Erlang version, or dependencies (based on the hash of `mix.lock`). I will keep `restore-keys` in mind if the current approach is found to be too conservative.